### ISSUE TITLE
Question about offline dataset

### ISSUE THREAD

**Issue by SamorsetTuska** [yaolinli/TimeChat-Online#24]
Hello, thank you for your excellent work! I encountered some issues while extracting offline video frames according to the annotations you provided. For instance, "Tarsier2-Recap-585K/Video-10M/2487878.mp4" is a 10-second short video, but there is "tarsier2/Video-10M/frames/2_6/2487878.mp4/000247.jpg" in the annotations. Another example is that Tarsier-Recap-585K contains Kinetics-700, while the annotations use Kinetics-700_2020. Would you be willing to provide the frames you extracted?

**Comment by jadmax0w0** [comment]
I encountered the same problem :'( the way I fixed this is that I simply uniformly re-sampled the entire video using only the total frames number calculated from the annotation file, regardless of how many frames the frame contains, and of what frame indices are in the annotation file. I dont know whether the result on dataset processed in this way could be consistent with the original work, still experimenting :(

**Comment by yaolinli** [comment]
Thanks a lot for reporting this and for the clear examples! 
To make training more convenient, we will upload pre-extracted offline frames to our Hugging Face dataset link.
**I’ll update here with the Hugging Face link once the upload is live**. Thanks again for your patience and for helping us improve the release.

**Comment by SamorsetTuska** [comment]
> Thanks a lot for reporting this and for the clear examples! To make training more convenient, we will upload pre-extracted offline frames to our Hugging Face dataset link. **I’ll update here with the Hugging Face link once the upload is live**. Thanks again for your patience and for helping us improve the release.

Thank you. Additionally, have you conducted any experiments using only the TimeChat-Online-139K dataset? The results of my reproduction are 72.79 on StreamingBench and 46.41 on OVO-Bench, with both experiments utilizing the default training and evaluation parameters. I am unsure whether these results are within the normal range.

**Comment by yaolinli** [comment]
Yes. In our experiment trained only on TimeChat-Online-139K, the results on StreamingBench (Real-Time Visual Understanding) are as follows:

**Dropping 44.1% of tokens**: 73.2% accuracy
**Dropping 82.6% of tokens**: 71.6% accuracy

The result in  paper Table 1 is trained with mixed data.

<img width="1223" height="699" alt="Image" src="https://github.com/user-attachments/assets/d91741f9-bda1-4dfd-a176-fe6ccd9affd7" />

**Comment by SamorsetTuska** [comment]
Thank you for sharing the results! I used LoRA for training instead of full fine-tuning, so the result should be within the normal range. Additionally, could you please let me know when the complete dataset will be released?

**Comment by yahooo-m** [comment]
Any update about the dataset?

**Comment by yaolinli** [comment]
# Update on Offline Dataset Frames

Thanks for your question, and sorry for the delay regarding the offline datasets.

### R1. The offline video frames are not available
Some of the general offline data were originally stored on our shared dataset root. Due to space management, these files were later moved or cleaned up by other users. As a result, **we can no longer fully recover the original extracted frames**, and certain frame indices in the annotations may not match the currently available files.

### R2. How we handled frames during training
In our training pipeline, for offline datasets, we always used a **fixed 64-frame input**, regardless of the actual video length:

1. Extract all frames according to the video’s FPS (e.g. 25.0 FPS)
2. Uniformly sample 64 frames as the model input (as shown in the annotation files)

Therefore, the exact frame numbers listed in the annotations are *not required* to reproduce our training data.

### R3. Recommended way to reproduce the offline dataset
Please download the raw videos directly from the official dataset sources and extract frames on your own:

1. Download videos from the dataset repositories  
2. Extract frames using the original video FPS  like 25.0
3. Uniformly sample 64 frames (or any fixed number your setup supports)  
4. Combine them with the TimeChat-Online dataset for training  

This will yield results consistent with our training pipeline.

### Official offline dataset links
- **VideoChat-Flash**  
  https://huggingface.co/datasets/OpenGVLab/VideoChat-Flash-Training-Data  

- **Tarsier2-Recap-585K**  
  https://huggingface.co/datasets/omni-research/Tarsier2-Recap-585K  

- **LLaVA-Video-178K**  
  https://huggingface.co/datasets/lmms-lab/LLaVA-Video-178K  

Using a consistent uniform-sampling strategy (e.g., fixed 64 frames) will align your data processing with ours.
